packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialog.js

```diff
@@ -162,7 +163,12 @@ const mergeProps = (stateProps, { dispatch, ...dispatchProps }, ownProps) => ({
     // constrain the fetch by 1st layer date range
     if (periodGranularity) {
       defaultStartDate = startDate;
-      defaultEndDate = endDate;
+
+      // set the endDate to be end of the startDate period
```

Comment:
Set the startDate to the start of the startDate period as well, and I think you've handled all the edge cases, and can close that other ticket!

It might be good to align it with the `getDefaultDates` function too though.

If this is getting too much work, tbh I'd remove this handling of one edge case in favour of doing them all properly down the track. We can just say that for now, the drill downs either have to have the same period granularity as the parent, or no period granularity.

Reply:
Sorry I forgot to address this comment before merging. I was a bit unsure how to proceed as Klaus was most of the way through testing when we raised this question. I think the best thing is to leave it as is with the main case working and a couple of the most likely edge cases handled as it has been tested and works well and then keep the other task to come back and fix it up fully when we have time. I hope that is ok? I have added some helpful notes to the supplementary task [here](https://github.com/beyondessential/tupaia-backlog/issues/1238)

Reply:
Yep all good!